### ISSUE TITLE
E_COMPILE_ERROR - Declaration of Grav\Plugin\Login\OAuth2\Providers\GitlabProvider::initProvider(array $options) must be compatible with Grav\Plugin\Login\OAuth2\Providers\BaseProvider::initProvider(array $options): void

### ISSUE THREAD

**Issue by Ynn** [trilbymedia/grav-plugin-login-oauth2-extras#4]
Hi,

I had an issue with GitlabProvider : 

```
E_COMPILE_ERROR - Declaration of Grav\Plugin\Login\OAuth2\Providers\GitlabProvider::initProvider(array $options) must be compatible with Grav\Plugin\Login\OAuth2\Providers\BaseProvider::initProvider(array $options): void
```

I fixed the problem by adding `void` in the function signature : 

```php
class GitlabProvider extends ExtraProvider
{
    protected $name = 'Gitlab';
    protected $classname = 'Omines\\OAuth2\\Client\\Provider\\Gitlab';

    public function initProvider(array $options) : void
``` 

I don't know if it depends on the version of PHP used. For information i am using PHP Version 8.0.6

**Comment by w00fz** [comment]
Yes that’s because of php 8. Could you create a PR for this?

**Comment by khanduras** [comment]
It's happening with me in with PHP 7.3 as well

```Whoops\Exception\ErrorException Declaration of Grav\Plugin\Login\OAuth2\Providers\AzureProvider::initProvider(array $options) must be compatible with Grav\Plugin\Login\OAuth2\Providers\BaseProvider::initProvider(array $options): void```

**Comment by rhukster** [comment]
These should all be fixed with latest 2.1.2 release
